.palantir/revapi.yml

```diff
@@ -1201,6 +1201,33 @@ acceptedBreaks:
       old: "method void org.apache.iceberg.MetadataUpdate.SetStatistics::<init>(long,\
         \ org.apache.iceberg.StatisticsFile)"
       justification: "Removing deprecated code"
+    org.apache.iceberg:iceberg-data:
+    - code: "java.method.numberOfParametersChanged"
+      old: "method void org.apache.iceberg.data.GenericAppenderFactory::<init>(org.apache.iceberg.Schema,\
+        \ org.apache.iceberg.PartitionSpec, int[], org.apache.iceberg.Schema, org.apache.iceberg.Schema)"
+      new: "method void org.apache.iceberg.data.GenericAppenderFactory::<init>(org.apache.iceberg.Table,\
+        \ int[], org.apache.iceberg.Schema, org.apache.iceberg.Schema)"
+      justification: "The previous API should be deprecated because, by default, it\
+        \ prevents table properties from being used by the metric config."
+    - code: "java.method.parameterTypeChanged"
+      old: "parameter void org.apache.iceberg.data.GenericAppenderFactory::<init>(===org.apache.iceberg.Schema===)"
+      new: "parameter void org.apache.iceberg.data.GenericAppenderFactory::<init>(===org.apache.iceberg.Table===)"
+      justification: "The previous API should be deprecated because, by default, it\
+        \ prevents table properties from being used by the metric config."
+    - code: "java.method.parameterTypeChanged"
+      old: "parameter void org.apache.iceberg.data.GenericAppenderFactory::<init>(===org.apache.iceberg.Schema===,\
+        \ org.apache.iceberg.PartitionSpec)"
+      new: "parameter void org.apache.iceberg.data.GenericAppenderFactory::<init>(===org.apache.iceberg.Table===,\
+        \ org.apache.iceberg.Schema)"
+      justification: "The previous API should be deprecated because, by default, it\
+        \ prevents table properties from being used by the metric config."
+    - code: "java.method.parameterTypeChanged"
+      old: "parameter void org.apache.iceberg.data.GenericAppenderFactory::<init>(org.apache.iceberg.Schema,\
+        \ ===org.apache.iceberg.PartitionSpec===)"
+      new: "parameter void org.apache.iceberg.data.GenericAppenderFactory::<init>(org.apache.iceberg.Table,\
+        \ ===org.apache.iceberg.Schema===)"
+      justification: "The previous API should be deprecated because, by default, it\
+        \ prevents table properties from being used by the metric config."
```

Comment:
These are always bad. Could you make sure that the API is not changed? Maybe use some default values and keep the old constructors?

Reply:
OK, I can pass a null table value if necessary